WalletWasabi.Tests/NodeBuilding/NodeBuilder.cs

```diff
@@ -11,202 +11,61 @@
-		public NodeConfigParameters ConfigParameters { get; } = new NodeConfigParameters();
-		public Network Network => Network.RegTest;
-
-		public async Task<CoreNode> CreateNodeAsync(bool start = false)
-		{
-			var child = Path.Combine(Root, Last.ToString());
-			Last++;
-			try
-			{
-				var cfgPath = Path.Combine(child, "data", "bitcoin.conf");
-				if (File.Exists(cfgPath))
-				{
-					var config = NodeConfigParameters.Load(cfgPath);
-					var rpcPort = config["regtest.rpcport"];
-					var rpcUser = config["regtest.rpcuser"];
-					var rpcPassword = config["regtest.rpcpassword"];
-					var pidFileName = config["regtest.pid"];
-					var credentials = new NetworkCredential(rpcUser, rpcPassword);
-					try
+					var pidFile = Path.Combine(dataDir, "regtest", pidFileName);
+					if (File.Exists(pidFile))
 					{
-						var rpc = new RPCClient(credentials, new Uri("http://127.0.0.1:" + rpcPort + "/"), Network.RegTest);
-						await rpc.StopAsync();
-
-						var pidFile = Path.Combine(child, "data", "regtest", pidFileName);
-						if (File.Exists(pidFile))
-						{
-							var pid = File.ReadAllText(pidFile);
-							using var process = Process.GetProcessById(int.Parse(pid));
-							process.WaitForExit();
-						}
-						else
-						{
-							var allProcesses = Process.GetProcesses();
-							var bitcoindProcesses = allProcesses.Where(x => x.ProcessName.Contains("bitcoind"));
-							if (bitcoindProcesses.Count() == 1)
-							{
-								var bitcoind = bitcoindProcesses.First();
-								bitcoind.WaitForExit();
-							}
-						}
+						var pid = await File.ReadAllTextAsync(pidFile);
+						using var process = Process.GetProcessById(int.Parse(pid));
+						await process.WaitForExitAsync(CancellationToken.None);
 					}
-					catch (Exception)
+					else
 					{
+						var allProcesses = Process.GetProcesses();
+						var bitcoindProcesses = allProcesses.Where(x => x.ProcessName.Contains("bitcoind"));
+						if (bitcoindProcesses.Count() == 1)
+						{
+							var bitcoind = bitcoindProcesses.First();
+							await bitcoind.WaitForExitAsync(CancellationToken.None);
+						}
 					}
 				}
-				await IoHelpers.DeleteRecursivelyWithMagicDustAsync(child);
-				await TryRemoveWorkingDirectoryAsync();
-				Directory.CreateDirectory(WorkingDirectory);
-			}
-			catch (DirectoryNotFoundException)
-			{
-			}
-			var node = await CoreNode.CreateAsync(child, this);
-			Nodes.Add(node);
-			if (start)
-			{
-				await node.StartAsync();
-			}
-			return node;
-		}
-
-		public Task StartAllAsync()
-		{
-			var startNodesTaskList = Nodes
-				.Where(n => n.State == CoreNodeState.Stopped)
-				.Select(n => n.StartAsync()).ToArray();
-			return Task.WhenAll(startNodesTaskList);
-		}
-
-		public void Dispose()
-		{
-			foreach (CoreNode node in Nodes)
-			{
-				node?.TryKillAsync().GetAwaiter().GetResult();
-			}
-
-			foreach (IDisposable disposable in Disposables)
-			{
-				disposable?.Dispose();
+				catch (Exception)
+				{
```

Comment:
Is this catch supposed to be empty?

Reply:
No, this smells, but it's copypaste, so it doesn't change behavior.